src/main/java/org/opentripplanner/routing/impl/PathFinder/WalkToBikePathFinder.java

```diff
@@ -0,0 +1,162 @@
+package org.opentripplanner.routing.impl.PathFinder;
+
+import org.opentripplanner.api.resource.DebugOutput;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.TraverseModeSet;
+import org.opentripplanner.routing.error.PathNotFoundException;
+import org.opentripplanner.routing.error.VertexNotFoundException;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.spt.GraphPath;
+import org.opentripplanner.standalone.Router;
+import java.util.*;
+
+public class WalkToBikePathFinder extends PathFinder {
+
+    public WalkToBikePathFinder(Router router) { super(router); }
+
+    /* Try to find N paths through the Graph */
+    public List<GraphPath> graphPathFinderEntryPoint (RoutingRequest request) {
+
+        // We used to perform a protective clone of the RoutingRequest here.
+        // There is no reason to do this if we don't modify the request.
+        // Any code that changes them should be performing the copy!
+
```

Comment:
This method is mostly repeated in all classes. Shouldn't it be in the superclass and simply extended in each child?